test/e2e/data/infrastructure-metal3/bases/centos-kubeadm-config/centos-kubeadm-config.yaml

```diff
@@ -126,7 +126,11 @@ spec:
           container-runtime-endpoint: unix:///var/run/crio/crio.sock
           feature-gates: AllAlpha=false
           node-labels: metal3.io/uuid={{ ds.meta_data.uuid }}
-          provider-id: metal3://{{ ds.meta_data.uuid }}
+{% if CAPM3_VERSION == "v1alpha5" %}
+          provider-id: "metal3://{{ '{{ ds.meta_data.uuid }}' }}"
+{% else %}
+          provider-id: "metal3://{{ '{{ ds.meta_data.providerid }}' }}"
+{% endif %}
```

Comment:
I do not think the `if else` will be interpreted here, else I would suggest to have the logic somewhere in the code and exports the results to a unique var that should be replaced here just like we are doing with `capi_version` for example